aten/src/ATen/native/MaxPooling.cpp

```diff
@@ -126,9 +148,11 @@ Tensor max_pool1d(
       !self.device().is_cpu() ||
       isTensorSubclassLike(self)) {
     // Needs indices for grad and with_indices defines CUDA dispatch
+    check_max_pool1d(self, kernel_size, stride, padding, dilation);
```

Comment:
bring `check_max_pool1d` to above the conditional, as you are calling ti in all cases.